TypeCobol.LanguageServer/WorkspaceProject.cs

```diff
@@ -0,0 +1,202 @@
+        /// <param name="analyzerProviderWrapper">The new Analyzer provider if any, null otherwise</param>
+        /// <param name="copyFolders">The new list of copy folders associated to this project if any, null otherwise</param>
+        public void Configure(DocumentFormat format, TypeCobolOptions options, AnalyzerProviderWrapper analyzerProviderWrapper, List<string> copyFolders)
+        {
+            bool updateFormat = false;
+            var newFormat = Project.Format;
+            if (format != null)
+            {
+                updateFormat = true;
+                newFormat = format;
+            }
+
+            bool updateOptions = false;
+            var newOptions = Project.CompilationOptions;
+            if (options != null)
+            {
+                updateOptions = true;
+                newOptions = options;
+            }
+
+            bool updateAnalyzerProvider = false;
+            var newAnalyzerProvider = Project.AnalyzerProvider;
+            if (analyzerProviderWrapper != null)
+            {
+                updateAnalyzerProvider = true;
+                newAnalyzerProvider = analyzerProviderWrapper;
+            }
+
+            var oldCopyFolders = GetCopyFolders();
+            bool updateCopyFolders = false;
+            var newCopyFolders = oldCopyFolders.ToList();
+            if (copyFolders != null && AreCopyFoldersSame(oldCopyFolders, copyFolders))
+            {
+                updateCopyFolders = true;
+                newCopyFolders = copyFolders;
+            }
+
+            if (updateFormat || updateOptions || updateAnalyzerProvider || updateCopyFolders)
+            {
+                var oldCompilationProject = Project;
+                var newCompilationProject = new CompilationProject(oldCompilationProject.Name, oldCompilationProject.RootDirectory, Helpers.DEFAULT_EXTENSIONS, newFormat, newOptions, newAnalyzerProvider);
+                foreach (var newCopyFolder in newCopyFolders)
+                {
+                    newCompilationProject.SourceFileProvider.AddLocalDirectoryLibrary(newCopyFolder, false, Helpers.DEFAULT_COPY_EXTENSIONS, newFormat);
+                }
+
+                Project = newCompilationProject;
+                // Change the target CompilationProject instance for each document.
+                foreach (var docCtx in _openedDocuments.Values)
+                {
+                    docCtx.FileCompiler.CompilationProject = Project;
+                }
+                _workspace.ScheduleRefresh(this, updateCopyFolders);
+            }
+        }
+
+        /// <summary>
+        /// Indicates whether this workspace project has opened documents or not.
+        /// </summary>
+        internal bool IsEmpty => _openedDocuments.IsEmpty;
+
+        /// <summary>
+        /// Refresh all documents of this WorkspaceProject instance
+        /// </summary>
+        internal void RefreshOpenedFiles()
```

Comment:
Not needed, the refresh is triggered by the WorkspaceProject itself

Reply:
As discussed for the moment it is used.

Reply:
done in [50aad65](https://github.com/TypeCobolTeam/TypeCobol/pull/2158/commits/50aad651b088759acfc704aa689a84ee67050e2d)